### PR TITLE
Handle allocation fails in GPS UAVCAN

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

**Comment by davidbuzz** [comment]
Isn't it generally considered ok for a new that fails and returns a nullptr to be tested as a boolean with a ternary or coalescing operator ...?
auto c = a ?: b;

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_GPS/AP_GPS_UAVCAN.cpp [hunk 1]
```diff
@@ -103,6 +103,10 @@ void AP_GPS_UAVCAN::subscribe_msgs(AP_UAVCAN* ap_uavcan)
 
     uavcan::Subscriber<uavcan::equipment::gnss::Fix, FixCb> *gnss_fix;
     gnss_fix = new uavcan::Subscriber<uavcan::equipment::gnss::Fix, FixCb>(*node);
+    if (gnss_fix == nullptr) {
+        AP_HAL::panic("AP_GPS_UAVCAN: Unable to allocate memory for gnss_fix subscriber");
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_UAVCAN.cpp:107]: a wrapper around config_error() called allocation_error() that just takes a string for the item being allocated would save flash, and allow users to disable the failing subsystem

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_BoardConfig/AP_BoardConfig.cpp [hunk 2]
```diff
@@ -394,6 +394,43 @@ void AP_BoardConfig::config_error(const char *fmt, ...)
     }
 }
 
+void AP_BoardConfig::allocation_error(const char *fmt, ...)
```
Comment (tridge) [inline libraries/AP_BoardConfig/AP_BoardConfig.cpp:397]: better to have a common method _config_error() that takes a const char *prefix

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Param/AP_Param.cpp [hunk 3]
```diff
@@ -2483,7 +2483,10 @@ void AP_Param::set_defaults_from_table(const struct defaults_table_struct *table
         if (!AP_Param::set_default_by_name(table[i].name, table[i].value)) {
             char *buf = nullptr;
             if (asprintf(&buf, "param deflt fail:%s", table[i].name) > 0) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
```
Comment (tridge) [inline libraries/AP_Param/AP_Param.cpp:2487]: should be "%s", buf

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by peterbarker — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**